tests: net: socket: getnameinfo: move to new ztest API

hunterhu3000 opened the pull request:
Move getnameinfo tests to use new ztest API
TEST=twister -T tests/net/socket/getnameinfo

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented on tests/net/socket/getnameinfo/src/main.c:
```diff
@@ -12,7 +12,7 @@ LOG_MODULE_REGISTER(net_test, CONFIG_NET_SOCKETS_LOG_LEVEL);
 #include <zephyr/net/socket.h>
 
-void test_getnameinfo_ipv4(void)
+ZTEST_USER(socket_getnameinfo_test_suite, test_getnameinfo_ipv4)
```
no need for _test_suite here.

hunterhu3000 replied:
Updated. Thanks to review.

hunterhu3000 committed:
tests: net: socket: getnameinfo: move to new ztest API

Move getnameinfo tests to use new ztest API
TEST=twister -T tests/net/socket/getnameinfo

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>
